airbyte-integrations/connectors/source-amazon-seller-partner/unit_tests/test_client.py

```diff
@@ -0,0 +1,111 @@
+from source_amazon_seller_partner.client import BaseClient
+
+SP_CREDENTIALS = {
+    "refresh_token": "ABC",
+    "lwa_app_id": "lwa_app_id",
+    "lwa_client_secret": "lwa_client_secret",
+    "aws_access_key": "aws_access_key",
+    "aws_secret_key": "aws_secret_key",
+    "role_arn": "role_arn",
+    "start_date": "start_date",
+}
+
+GET_FLAT_FILE_ALL_ORDERS_DATA_BY_ORDER_DATE_GENERAL = "GET_FLAT_FILE_ALL_ORDERS_DATA_BY_ORDER_DATE_GENERAL"
+ORDERS = "Orders"
+_ENTITIES = [GET_FLAT_FILE_ALL_ORDERS_DATA_BY_ORDER_DATE_GENERAL, ORDERS]
+
+MARKETPLACE = "India"
+ORDERS_RESPONSE = [{"orderId": 1}]
+
+
+class MockAmazonClient:
+    def __init__(self, credentials, marketplace):
+        self.credentials = credentials
+        self.marketplace = marketplace
+
+    def fetch_orders(updated_after, page_count, next_token=None):
+        return ORDERS_RESPONSE
+
+
+def test_check_connection(mocker):
+    mocker.patch("source_amazon_seller_partner.client.AmazonClient",
+                 return_value=MockAmazonClient)
+    base_client = BaseClient(
+        role_arn=SP_CREDENTIALS["role_arn"], start_date=SP_CREDENTIALS["start_date"],
+        refresh_token=SP_CREDENTIALS["refresh_token"], lwa_app_id=SP_CREDENTIALS["lwa_app_id"],
+        lwa_client_secret=SP_CREDENTIALS["lwa_client_secret"], aws_secret_key=SP_CREDENTIALS["aws_secret_key"],
+        aws_access_key=SP_CREDENTIALS["aws_access_key"], marketplace=MARKETPLACE)
+
+    assert ORDERS_RESPONSE == base_client.check_connection()
+
+
+def test_get_cursor_state():
+    cursor_value = "2021-05-02"
+    end_date = "2021-06-02"
+
+    base_client = BaseClient(
+        role_arn=SP_CREDENTIALS["role_arn"], start_date=SP_CREDENTIALS["start_date"],
+        refresh_token=SP_CREDENTIALS["refresh_token"], lwa_app_id=SP_CREDENTIALS["lwa_app_id"],
+        lwa_client_secret=SP_CREDENTIALS["lwa_client_secret"], aws_secret_key=SP_CREDENTIALS["aws_secret_key"],
+        aws_access_key=SP_CREDENTIALS["aws_access_key"], marketplace=MARKETPLACE)
+
+    assert "2021-05-03" == base_client._get_cursor_state(
+        cursor_value=cursor_value, end_date=end_date)
+
+    cursor_value = "2021-02-28"
+    end_date = "2021-03-02"
+
+    assert "2021-03-03" == base_client._get_cursor_state(
+        cursor_value=cursor_value, end_date=end_date)
+
+
+def test_get_records():
+    data = {"document": "name\ttest\nairbyte\t1"}
+    base_client = BaseClient(
+        role_arn=SP_CREDENTIALS["role_arn"], start_date=SP_CREDENTIALS["start_date"],
+        refresh_token=SP_CREDENTIALS["refresh_token"], lwa_app_id=SP_CREDENTIALS["lwa_app_id"],
+        lwa_client_secret=SP_CREDENTIALS["lwa_client_secret"], aws_secret_key=SP_CREDENTIALS["aws_secret_key"],
+        aws_access_key=SP_CREDENTIALS["aws_access_key"], marketplace=MARKETPLACE)
+
+    assert [{"name": "airbyte", "test": "1"}] == base_client._get_records(data)
+
+
+def test_apply_conversion_window():
+    current_date = "2021-03-04"
+    base_client = BaseClient(
```

Comment:
would it make sense to refactor this initialization of `BaseClient` into a helper method since we seem to be doing it a lot?